### PR TITLE
fixes for redhat internal tasks

### PR DESCRIPTION
**Pull request by rakeshgm**
added couple of the internal tasks for redhat that were missing and fixes for the same. 

Signed-off-by: rakeshgm <rakeshgm@redhat.com>

### CODE REVIEW

File: teuthology/task/internal/__init__.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 from teuthology.exceptions import VersionNotFoundError
 from teuthology.job_status import get_status, set_status
 from teuthology.orchestra import cluster, remote, run
-
+from .redhat import setup_cdn_repo, setup_base_repo, setup_additional_repo, setup_stage_cdn # noqa
```
Comment (kshtsk) [inline teuthology/task/internal/__init__.py:22]: Could you please use full path to the redhat module?

Comment (kshtsk) [reply teuthology/task/internal/__init__.py:22]: And remove # noqa

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
is this needed for anything else but ceph 'master'?

**Comment by vasukulkarni** [comment]
> is this needed for anything else but ceph 'master'?

I am not sure i understand your question,  the task allows to run ceph in octo lab

File: teuthology/task/internal/__init__.py (hunk 1 above)
Comment (vasukulkarni) [reply teuthology/task/internal/__init__.py:22]: @rakeshgm can you see if the test passes with noqa and full path?

Comment (kshtsk) [reply teuthology/task/internal/__init__.py:22]: @vasukulkarni I would prefer to follow one style in imports.

Comment (kshtsk) [reply teuthology/task/internal/__init__.py:22]: also in some circumstances it can fail

**Comment by kshtsk** [comment]
> > is this needed for anything else but ceph 'master'?
> 
> I am not sure i understand your question, the task allows to run ceph in octo lab

What is 'octo' lab? The question is the committed code supposed to be run against ceph branches which are not switched to py3, if so the is probably need to backport them to py2 branch.

File: teuthology/task/internal/__init__.py (hunk 1 above)
Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: Done. relative import added

Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: test passes with full path.

Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: tests failed because flake8 reported unused imports

Comment (kshtsk) [reply teuthology/task/internal/__init__.py]: so why this imports needed if they are not used?

Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: they are actually being called from run.py https://github.com/ceph/teuthology/blob/master/teuthology/run.py#L236.

Comment (kshtsk) [reply teuthology/task/internal/__init__.py]: > they are actually being called from run.py https://github.com/ceph/teuthology/blob/master/teuthology/run.py#L236.

@rakeshgm the code you're referring is not used for calling them, it is just compose the task list, the modules for them are loaded automatically in other place, see as an example other jobs in the task list are not imported directly here in this module.

Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: @kshtsk , I intended to mention the same. sorry I did not phrase it correctly.

**Comment by kshtsk** [comment]
@susebot run deploy

File: teuthology/task/internal/__init__.py (hunk 1 above)
Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: @kshtsk  so to fix the flake8 errors, I can move the tasks from `redhat` and add them in `__init__` which might fix the problem else we could check for flake8 options to ignore such errors i.e if you think its a good idea. 
Or any other suggestions.

Comment (kshtsk) [reply teuthology/task/internal/__init__.py]: @rakeshgm no, just remove this import line, and that's it.

Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: `setup_cdn_repo, setup_base_repo, setup_additional_repo, setup_stage_cdn` tasks are present in `redhat.py` unlike others in `__init__` for example in run.py we have somthing like this `{'internal.setup_cdn_repo': None}` .
importing this line gets the task.

Comment (kshtsk) [reply teuthology/task/internal/__init__.py]: Ah, I see, why not using `{'internal.redhat.setup_cdn_repo': None}` then?

Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: I tried that a while ago. the split here fails. 
https://github.com/ceph/teuthology/blob/master/teuthology/run_tasks.py#L21

Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: I pushed a small [change here](6261eae46675eacc515f00bc4f3f1b070dbeb85b)

Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: I pushed a small change [here](https://github.com/ceph/teuthology/pull/1479/commits/6261eae46675eacc515f00bc4f3f1b070dbeb85b) to this PR.

Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: looks like the tests passed.

Comment (kshtsk) [reply teuthology/task/internal/__init__.py]: > I tried that a while ago. the split here fails.
> https://github.com/ceph/teuthology/blob/master/teuthology/run_tasks.py#L21
> returns more than 2.

Oh... shoot, it does not support submodules then... Seems like redhat is the first ever module is trying to do so.
I guess we could add a support for submodules.

Comment (kshtsk) [reply teuthology/task/internal/__init__.py]: > I pushed a small change [here](https://github.com/ceph/teuthology/pull/1479/commits/6261eae46675eacc515f00bc4f3f1b070dbeb85b) to this PR.

That's too hacky, I would go with `# noqa` tag accompanying a comment why it is needed?

Comment (kshtsk) [reply teuthology/task/internal/__init__.py]: however... it's fine....

Comment (kshtsk) [reply teuthology/task/internal/__init__.py]: we can add support for submodules later...

**Comment by kshtsk** [comment]
@susebot run deploy

File: teuthology/task/internal/__init__.py (hunk 1 above)
Comment (vasukulkarni) [reply teuthology/task/internal/__init__.py]: yeah the split and support for single module was the reason I think we did that, I suggest we keep the original code and add comment and move on until the support exists in teuthology.

**Comment by vasukulkarni** [comment]
> > > is this needed for anything else but ceph 'master'?
> > 
> > 
> > I am not sure i understand your question, the task allows to run ceph in octo lab
> 
> What is 'octo' lab? The question is the committed code supposed to be run against ceph branches which are not switched to py3, if so the is probably need to backport them to py2 branch.

We dont need py2 support, I didnt read your question correctly but octo is internal lab and you can ignore my previous answer :)

File: teuthology/task/internal/__init__.py (hunk 1 above)
Comment (rakeshgm) [reply teuthology/task/internal/__init__.py]: reverted, add full path import along with `noqa`

**Comment by vasukulkarni** [comment]
@kshtsk do you have any more concerns here?

File: teuthology/task/internal/__init__.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 from teuthology.exceptions import VersionNotFoundError
 from teuthology.job_status import get_status, set_status
 from teuthology.orchestra import cluster, remote, run
+from teuthology.task.internal.redhat import setup_cdn_repo, setup_base_repo, setup_additional_repo, setup_stage_cdn  # noqa
```
Comment (kshtsk) [inline teuthology/task/internal/__init__.py:22]: so could you please a comment here why this import need here? also maybe add a `todo` tag for adding support of submodules

Comment (vasukulkarni) [reply teuthology/task/internal/__init__.py:22]: yeah good point, @rakeshgm can you fix that

**Comment by kshtsk** [comment]
> @kshtsk do you have any more concerns here?

I'm fine with it, but a comment about the need of the import

**Comment by kshtsk** [comment]
@susebot run deploy

File: teuthology/task/internal/__init__.py (hunk 2 above)
Comment (rakeshgm) [reply teuthology/task/internal/__init__.py:23]: done. added comment and todo

**Pull request merged**